primitives/src/rescue/sponge.rs

```diff
@@ -0,0 +1,441 @@
+// Copyright (c) 2022 Espresso Systems (espressosys.com)
+// This file is part of the Jellyfish library.
+
+// You should have received a copy of the MIT License
+// along with the Jellyfish library. If not, see <https://mit-license.org/>.
+
+//! This file contains the APIs wrappers for ark-sponge
+
+use ark_ff::{BigInteger, PrimeField};
+use ark_sponge::{
+    Absorb, CryptographicSponge, FieldBasedCryptographicSponge, FieldElementSize, SpongeExt,
+};
+use ark_std::{string::ToString, vec, vec::Vec};
+use jf_utils::{field_switching, pad_with_zeros};
+use num_bigint::BigUint;
+
+use super::{errors::RescueError, Permutation, RescueParameter, RescueVector, RATE, STATE_SIZE};
+
+#[derive(Clone, Default)]
+/// A rescue hash function consists of a permutation function and
+/// an internal state.
+pub struct RescueSponge<F: RescueParameter, const CHUNK_SIZE: usize> {
+    pub(crate) state: RescueVector<F>,
+}
+
+/// CRHF
+pub type RescueCRH<F> = RescueSponge<F, RATE>;
```

Comment:
Sorry to be picky. I felt like, `RescueCRHF` is a better name here.
WDYT @chancharles92 @alxiong

Reply:
hahah, sorry, I suggested the `CRH` name, probably because I'm influenced by the [arkwork's naming](https://github.com/arkworks-rs/crypto-primitives/tree/main/src/crh) convention and Aleo's snarkVM's naming. 

I'm okay with both, and my apology to @tessico first in case we change this back haha. 😜

Reply:
I'm also neutral about this one. As seemingly there's preference for `CRHF`, I renamed it in the [latest commit](https://github.com/EspressoSystems/jellyfish/pull/146/commits/ae2bbe1b3a8de6ac7331e0bdc1c01dbe3e2aa3ff).